src/storage/mod.rs

```diff
@@ -129,6 +140,145 @@ impl Storage {
         Ok(blob)
     }
 
+    pub(super) fn get_range(
+        &self,
+        path: &str,
+        max_size: usize,
+        range: FileRange,
+        compression: Option<CompressionAlgorithm>,
+    ) -> Result<Blob, Error> {
+        let mut blob = match &self.backend {
+            StorageBackend::Database(db) => db.get(path, max_size, Some(range)),
+            StorageBackend::S3(s3) => s3.get(path, max_size, Some(range)),
+        }?;
+        // file content encoding is ignored for ranges
+        // since we only have a range anyways, and we need the encoding
+        // for the range, not the file
```

Comment:
- `.get` uses the `Content-Encoding` header from S3 to determine the compression of the object
- in `.get_range`, even though the method looks similar, we are ignoring any content encoding that might come from S3, and only use the argument. 

Should I add more explanation? Or do you think it's no value and I should remove it?

Reply:
> we are ignoring any content encoding that might come from S3, and only use the argument.

Ah ok, this is what I missed. Now I'm confused: why would the argument be different from the encoding on S3?

Reply:
I probably will add a better comment explaining this when we came to a shared understanding. 

Until now the content-encoding on the normal file on S3 tells us which algorithm to use to decompress the binary stream so we get the original content back. 

When we have a zip-file the situation is like this: 
- the whole zip-file is its own format, so putting any content-encoding would be wrong, since you cannot use any compression algorithm to decompress the zip-file. 
- inside the zip-file we have a different situation: every file inside the archive can have a different compression algorithm. We stored that one in the index and need it here when we fetch the part of the zip-file that contains the binary stream of the single object we want.